styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java

```diff
@@ -97,10 +99,11 @@ void tick() {
 
       if (enabled.contains(workflow.id())) {
         try {
-          triggerListener.event(
+          CompletionStage<Void> processed = triggerListener.event(
               workflow,
               Trigger.natural(),
               decrementInstant(next, partitioning));
+          processed.toCompletableFuture().get(1, TimeUnit.MINUTES);
```

Comment:
What might cause timeout, a big queue to process? Is it possible the event eventually gets processed and job gets to run, but we might trigger it one more time?